keyvalue/clients/src/main/java/tachyon/client/keyvalue/LinearProbingIndex.java

```diff
@@ -79,9 +81,16 @@ public static LinearProbingIndex createEmptyIndex() {
    * @return an instance of linear probing index
    */
   public static LinearProbingIndex loadFromByteArray(ByteBuffer buffer) {
-    int numBuckets = buffer.limit() / BUCKET_SIZE_BYTES;
-    // TODO(binfan): fix the key count which is wrong now, see TACHYON-1555
-    return new LinearProbingIndex(buffer, numBuckets, 0);
+    int numBuckets = 0;
+    int keyCount = 0;
+    for (int offset = 0, limit = buffer.limit(); offset < limit; offset += BUCKET_SIZE_BYTES) {
```

Comment:
Since the limit is only used in the loop, this is more compact. Do we have code conventions about this?